### ISSUE TITLE
Ref on images

### ISSUE THREAD

**Issue by alex-r89** [Aljullu/react-lazy-load-image-component#32]
Im trying to manipulate an image that is passed into the LazyLoadImage component via a React Ref, however I dont seem to be able to do so. The reason being is because I would like to manipulate the image, for example zoom in (increase scale) on click.

It would be great if a ref could be passed into the LazyLoadImage component, which is then put onto the image.

I have tried putting a ref onto the LazyLoadImage component its self, but this doesn't seem to work and I am unable to manipulate the actual image (for example increase its scale).

Thanks,

**Comment by Aljullu** [comment]
Right, I imagine that's because `<LazyLoadImage>` uses a ref internally and that would overwrite any ref passed from above. I wonder if wrapping `<LazyLoadImage>` in a `<div>` or something similar would be a good enough solution in your case:
```JSX
  <div ref={this.imageRef}>
    <LazyLoadImage
      alt={image.alt}
      height={image.height}
      src={image.src} // use normal <img> attributes as props
      width={image.width} />
    <span>{image.caption}</span>
  </div>
```

**Comment by dantovbein** [comment]
Hi @Aljullu , what about passing refs to passing refs to placeholders? I'm figuring out that `ref.current` is always `null` *before* scrolling to the child element.

```
<LazyLoadComponent 
    threshold={200} 
    placeholder={<div ref={this.childRef}>Empty content :/</div>}>
        <div ref={this.itinerariesRef}>Child content!</div>
</LazyLoadComponent>
```
Is it possible?
Thanks in advance

**Comment by dzucconi** [comment]
I'm surprised this doesn't come up more. You need access to the `img` `completed` attribute to detect whether or not the image is loaded in a SSR context. You could forward the ref and internally compose using something like https://github.com/seznam/compose-react-refs

**Comment by jellohouse** [comment]
I would also like to have ref of the actual <img> inside LazyLoadImage! Please add this feature!
